src/github.com/getlantern/flashlight/client/reverseproxy.go

```diff
@@ -40,23 +28,25 @@ func (client *Client) initReverseProxy(bal *balancer.Balancer, dumpHeaders bool)
 		DisableKeepAlives: true,
 	}
 
+	// Just choose a random dialer that also takes care of things like the
+	// authentication token.
+	dialer := client.getBalancer().RandomDialer()
```

Comment:
This is the key change here. Just choose a random dialer from the balancer and use that instead of using the balancers's dial method. This allows us to know what dialer we're using ahead of time in order to set the auth token correctly.